spec/browsers/pandora_user_agent_spec.rb

```diff
@@ -1,147 +1,87 @@
 require 'user_agent'
 
-shared_examples_for 'Pandora' do
+shared_examples_for 'Pandora' do |version, platform, os, type|
   it "should return 'Pandora' as its browser" do
     expect(useragent.browser).to eq('Pandora')
   end
-end
-
-describe "Mozilla/5.0 (iPad; CPU OS 10_3_4 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Mobile/14G61 Pandora/1812.1" do
-  let!(:useragent) { UserAgent.parse("Mozilla/5.0 (iPad; CPU OS 10_3_4 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Mobile/14G61 Pandora/1812.1") }
 
-  it_should_behave_like "Pandora"
+  it "should return '#{version}' as its version" do
+    expect(useragent.version).to eq(version)
+  end
 
-  it "should return '1812.1' as its version" do
-    expect(useragent.version).to eq('1812.1')
+  it "should return '#{platform}' as its platform" do
+    expect(useragent.platform).to eq(platform)
   end
 
-  it "should return 'iPad' as its platform" do
-    expect(useragent.platform).to eq('iPad')
+  it "should return '#{os}' as its os" do
+    expect(useragent.os).to eq(os)
   end
 
-  it "should return 'iOS 10.3.4' as its os" do
-    expect(useragent.os).to eq('iOS 10.3.4')
+  if type == :mobile
+    it { expect(useragent).to be_mobile }
+  elsif type == :desktop
+    it { expect(useragent).to be_desktop }
+  elsif type == :bot
+    it { expect(useragent).to be_bot }
   end
+end
 
-  it { expect(useragent).to be_mobile }
+describe "Mozilla/5.0 (iPad; CPU OS 10_3_4 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Mobile/14G61 Pandora/1812.1" do
+  let!(:useragent) { UserAgent.parse("Mozilla/5.0 (iPad; CPU OS 10_3_4 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Mobile/14G61 Pandora/1812.1") }
+
+  it_should_behave_like 'Pandora', '1812.1', 'iPad', 'iOS 10.3.4', :mobile
 end
 
 describe "Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Mobile/14F89 Pandora/1901.1.1" do
   let!(:useragent) { UserAgent.parse("Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Mobile/14F89 Pandora/1901.1.1") }
 
-  it_should_behave_like "Pandora"
-
-  it "should return '1901.1.1' as its version" do
-    expect(useragent.version).to eq('1901.1.1')
-  end
-
-  it "should return 'iPhone' as its platform" do
-    expect(useragent.platform).to eq('iPhone')
-  end
-
-  it "should return 'iOS 10.3.2' as its os" do
-    expect(useragent.os).to eq('iOS 10.3.2')
-  end
-
-  it { expect(useragent).to be_mobile }
+  it_should_behave_like 'Pandora', '1901.1.1', 'iPhone', 'iOS 10.3.2', :mobile
```

Comment:
I like this version better with mobile as a symbol.  Versus a boolean flag in the arguments.  I think it reads better.  Can you make Overcast that way?